File: java/src/main/java/com/example/hexagonal/application/port/inbound/ProductPort.java
package com.example.hexagonal.application.port.inbound;

import com.example.hexagonal.application.port.dto.ProductDto;
import com.example.hexagonal.application.domain.Product;

public interface ProductPort {

  Product getProduct(Long productId);
  Product updateProduct(Long productId, ProductDto dto);
}

File: kotlin/src/main/kotlin/com/example/hexagonal/application/port/inbound/ProductPort.kt
package com.example.hexagonal.application.port.inbound

import com.example.hexagonal.application.domain.Product
import com.example.hexagonal.application.port.dto.ProductDto

interface ProductPort {
    fun getProduct(productId: Long): Product
    fun updateProduct(productId: Long, dto: ProductDto): Product
}
